recursive_art.py

```diff
@@ -20,9 +21,20 @@ def build_random_function(min_depth, max_depth):
         (See the assignment writ-eup for details on the representation of
         these functions)
     """
-    # TODO: implement this
-    pass
 
+    if min_depth<=1:
```

Comment:
These conditionals are a bit confusing - you could make them less so by adding an empty line between the separate statements (after lines 30 and 32) and commenting what exactly this portion of the code is doing.